sympy/solvers/solveset.py

```diff
@@ -943,7 +945,12 @@ def _solveset(f, symbol, domain, _check=False):
                     elif equation.has(Abs):
                         result += _solve_abs(f, symbol, domain)
                     else:
-                        result += _solve_as_rational(equation, symbol, domain)
+                        result_rational = _solve_as_rational(equation, symbol, domain)
+                        if isinstance(result_rational, ConditionSet):
+                            # may be a transcendental type equation
+                            result += _transolve(equation, symbol, domain)
```

Comment:
If you aren't going to implement solving in the complex domain then you should add the test here, perhaps:

```python
...
# may be a transcendental type equation
if not domain.is_subset(S.Reals):
    raise NotImplementedError('solving transcendent equation in the complex domain.')
result += _transolve(...
```

Reply:
Is getting a `ConditionSet` not a good idea?
As of now for the complex domain, `ConditionSet` is returned.

Reply:
oh, sorry. ConditionSet is fine.